lib/Service/InstanceService.php

```diff
@@ -36,33 +36,40 @@
 use OCA\Social\Exceptions\InstanceDoesNotExistException;
 use OCA\Social\Model\ActivityPub\ACore;
 use OCA\Social\Model\Instance;
+use OCP\IConfig;
 
 class InstanceService {
 	use TArrayTools;
 
-	private InstancesRequest$instancesRequest;
+	private InstancesRequest $instancesRequest;
 	private ConfigService $configService;
 	private MiscService $miscService;
+	private IConfig $config;
 
 	public function __construct(
-		InstancesRequest $instancesRequest, ConfigService $configService, MiscService $miscService
+		InstancesRequest $instancesRequest,
+		ConfigService $configService,
+		MiscService $miscService,
+		IConfig $config
 	) {
 		$this->instancesRequest = $instancesRequest;
 		$this->configService = $configService;
 		$this->miscService = $miscService;
+		$this->config = $config;
 	}
 
-
-	/**
-	 *
-	 */
-	public function createLocal(): void {
+	public function createLocal(): Instance {
+		$instance = new Instance();
+		$instance->setLocal(true)
+			->setVersion($this->config->getAppValue('social', 'installed_version', '0.0'))
+			->setApprovalRequired(false)
+			->setDescription($this->config->getAppValue('theming', 'slogan', 'a safe home for your data'))
+			->setTitle($this->config->getAppValue('theming', 'name', 'Nextcloud Social'));
+		$this->instancesRequest->save($instance);
+		return $instance;
```

Comment:
we need to implement on the front-end a way to assign a contactAccount to the `Instance`

Reply:
And also to add all the other fields. Could you create an issue so that we don't forget?

Reply:
done